packages/gatsby-plugin-page-creator/src/gatsby-node.ts

```diff
@@ -423,7 +423,7 @@ export async function onPluginInit(
   try {
     const pagesGlob = `**/**\\{*\\}**`
 
-    const files = await glob(pagesGlob, { cwd: pagesPath })
+    const files = await glob(pagesGlob, { cwd: pagesPath, onlyFiles: false })
```

Comment:
This doesn't seem to work as we would expect. Upon some local testing this seems to return paths to directories that match without including paths to files under those directories: For `a/{b.slug}/index.tsx` case it would just return `a/{b.slug}` instead of actual path to the template file we expect.

It seems to me that is caused more by glob pattern, and we shouldn't use `onlyFiles: false`. For example changing pattern to:

```diff
-const pagesGlob = `**/**\\{*\\}**`
+const pagesGlob = [`**/**\\{*\\}**/**`, `**/**\\{*\\}**`];
```

made it return what I would expect which is `a/{b.slug}/index.tsx`. It seems like current pattern only match if final segment in path has curly braces. Adding `/**` to the pattern make it match when there was curly braces in previous segments, but drops previous behaviour so it seems like those 2 patterns need to be combined? Or maybe there is syntax to have 1 pattern, but we didn't find one.

Given potential problems this could introduce I would suggest adding some tests for this:
1. Move lines 424-426 to separately utility function ( https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-page-creator/src/path-utils.ts is good place for it)
2. add some unit tests - you can use fixtures directory for test inputs

Reply:
Thanks for the feedback! I'll bring back my other version which fits more inline with what is asked here, complete with unit test and all!

Reply:
> It seems like current pattern only match if final segment in path has curly braces. Adding /** to the pattern make it match when there was curly braces in previous segments, but drops previous behaviour so it seems like those 2 patterns need to be combined? Or maybe there is syntax to have 1 pattern, but we didn't find one.

From what I tried, you can only do it with two basic patterns or use `()` from the advanced glob features to make a pattern (`**/**{**}**(/**`)). I'm not sure which of the two would be preferable.